packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -4990,19 +5042,24 @@ class RenderFollowerLayer extends RenderProxyBox {
 
   @override
   void paint(PaintingContext context, Offset offset) {
+    final Size? leaderSize = link.leaderSize;
+    final Offset effectiveLinkedOffset = leaderSize == null
```

Comment:
Could have asserted that `leaderSize != null || link.leader`, but that would break custom render objects that are pushing a `LeaderLayer` themselves (without using a `RenderLeaderLayer`), e.g, `RenderEditable`.